pkg/server/container_create_windows.go

```diff
@@ -517,18 +513,57 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
 		}
 	}
 
+	// hyperv pci devices are exposed as windows devices on the container spec
+	// given to hcsshim.
+	addOCIWindowsDevices(&g, rhcso, config.Devices)
+
 	return g.Config, nil
 }
 
-func addOCIWindowsDevices(g *generator, devs []*runtime.Device) {
+// getWindowsDeviceInfo is a helper function that returns a spec specified device's
+// prefix and device identifier. A prefix is any string before "://" in the spec
+// device's `HostPath`.
+func getWindowsDeviceInfo(hostPath string) (string, string) {
+	substrings := strings.SplitN(hostPath, "://", 2)
+	if len(substrings) <= 1 {
+		return "", ""
+	}
+	return substrings[0], substrings[1]
+}
+
+// validWindowsDeviceForIsolation is a helper function that returns if a given windows
+// device prefix is valid for the isolation type of the container.
+//
+// Hyper-V isolated containers only support the prefixes "gpu" and "vpci".
+//
+// Process isolated containers only support the prefixes "vpciLocationPath"
+// and "vpciClassGUID"
+//
+// Returns true if the device type is valid for the container, false otherwise
+func validWindowsDeviceForIsolation(prefix string, rhcso *runhcsoptions.Options) bool {
+	isolation := rhcso.SandboxIsolation
+	if isolation == runhcsoptions.Options_PROCESS {
+		return prefix == "vpciLocationPath" || prefix == "vpciClassGUID"
```

Comment:
In the past we've formatted these prefixes like `vpci-location-path` instead of `vpciLocationPath` (e.g.: `automanage-vhd://`).

Reply:
I wanted to keep the string used as a prefix (iow "vpciLocationPath") the same as the corresponding device ID type. If you don't think it's a problem to use -'s for both, then I'm okay changing it her and in the shim.

Reply:
Actually, I think that's fine in this class. I'll make the change.

Reply:
Didn't we define the name `vpciLocationPath` though? I thought it was just an identifier we picked, so we could change it to whatever we want.

Reply:
e.g. also change it in the shim